### PR TITLE
[LIBCLOUD-622] Support for config_drive in the Openstack driver

### PR DESCRIPTION
**Pull request by rnirmal**
Adds a 'ex_config_drive' option to create_node and ex_rebuild_node
for the Openstack driver.

### CODE REVIEW

File: libcloud/test/compute/fixtures/openstack_v1.1/_servers_26f7fbee_8ce1_4c28_887a_bfe8e4bb10fe.json [hunk 1]
```diff
@@ -40,7 +40,7 @@
                 }
             ]
         },
-        "config_drive": "",
+        "config_drive": "True",
```
Comment (Kami) [inline libcloud/test/compute/fixtures/openstack_v1.1/_servers_26f7fbee_8ce1_4c28_887a_bfe8e4bb10fe.json:43]: Hm, so a value is a string and not a boolean?

File: libcloud/compute/drivers/openstack.py [hunk 2]
```diff
@@ -1358,6 +1366,11 @@ def ex_rebuild(self, node, image, **kwargs):
                                     Can be either ``AUTO`` or ``MANUAL``.
         :type       ex_disk_config: ``str``
 
+        :keyword    ex_config_drive: Enables metadata injection in a server
```
Comment (Kami) [inline libcloud/compute/drivers/openstack.py]: Even if the OpenStack API takes a string and not a boolean, Libcloud API should take a boolean and this should get converted to the OpenStack format inside the method.

**Comment by Kami** [comment]
Thanks.

There is a minor issue with a string / boolean in a public API, but besides that it looks good.

**Comment by Kami** [comment]
It would also be nice if you could add some documentation on how to use `ex_config_drive` and `ex_userdata` :)

File: libcloud/test/compute/fixtures/openstack_v1.1/_servers_26f7fbee_8ce1_4c28_887a_bfe8e4bb10fe.json (hunk 1 above)
Comment (rnirmal) [inline libcloud/test/compute/fixtures/openstack_v1.1/_servers_26f7fbee_8ce1_4c28_887a_bfe8e4bb10fe.json:43]: Ah gotcha, yeah I was just mimicking what the api had, I'll get that changed to a boolean.

**Comment by rnirmal** [comment]
The value is still a string on the API side, but I've set it up as a boolean on the client side. I've also added an example to show how to use it.

**Comment by Kami** [comment]
Merged into trunk. Thanks.

**Pull request closed without merging**